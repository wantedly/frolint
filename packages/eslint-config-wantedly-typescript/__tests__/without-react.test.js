const ESLint = require("eslint").ESLint;
const baseConfig = require("../without-react");

const normalizePath = (path) => {
  return /node_modules/.test(path) ? path.split("node_modules")[1] : path;
};

describe("eslint-config-wantedly-typescript/without-react", () => {
  test("should match snapshot for", async () => {
    const config = await new ESLint({
      baseConfig,
      useEslintrc: false,
    }).calculateConfigForFile("test.ts");
    const keys = Object.keys(config);

    const normalizeRequiredKeys = ["extends", "parser"];
    normalizeRequiredKeys.forEach((key) => {
      const newConfig = config[key];

      if (Array.isArray(newConfig)) {
        config[key] = newConfig.map(normalizePath);
      } else {
        config[key] = normalizePath(newConfig);
      }
    });

    keys.forEach((key) => {
      expect(config[key]).toMatchSnapshot(key);
    });
  });
});
