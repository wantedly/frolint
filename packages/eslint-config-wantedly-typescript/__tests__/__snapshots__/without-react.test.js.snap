// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`eslint-config-wantedly-typescript/without-react should match snapshot for: env 1`] = `
{
  "browser": true,
  "es6": true,
  "jest/globals": true,
  "node": true,
}
`;

exports[`eslint-config-wantedly-typescript/without-react should match snapshot for: globals 1`] = `
{
  "flushPromises": true,
}
`;

exports[`eslint-config-wantedly-typescript/without-react should match snapshot for: ignorePatterns 1`] = `[]`;

exports[`eslint-config-wantedly-typescript/without-react should match snapshot for: noInlineConfig 1`] = `undefined`;

exports[`eslint-config-wantedly-typescript/without-react should match snapshot for: parser 1`] = `"/@typescript-eslint/parser/dist/index.js"`;

exports[`eslint-config-wantedly-typescript/without-react should match snapshot for: parserOptions 1`] = `
{
  "ecmaFeatures": {
    "experimentalObjectRestSpread": true,
    "jsx": true,
  },
  "sourceType": "module",
}
`;

exports[`eslint-config-wantedly-typescript/without-react should match snapshot for: plugins 1`] = `
[
  "use-macros",
  "@typescript-eslint",
  "jest",
  "jsx-a11y",
  "import",
]
`;

exports[`eslint-config-wantedly-typescript/without-react should match snapshot for: reportUnusedDisableDirectives 1`] = `undefined`;

exports[`eslint-config-wantedly-typescript/without-react should match snapshot for: rules 1`] = `
{
  "@babel/object-curly-spacing": [
    "off",
  ],
  "@babel/semi": [
    "off",
  ],
  "@typescript-eslint/adjacent-overload-signatures": [
    "error",
  ],
  "@typescript-eslint/array-type": [
    "off",
  ],
  "@typescript-eslint/ban-ts-comment": [
    "error",
  ],
  "@typescript-eslint/ban-tslint-comment": [
    "off",
  ],
  "@typescript-eslint/ban-types": [
    "error",
  ],
  "@typescript-eslint/block-spacing": [
    "off",
  ],
  "@typescript-eslint/brace-style": [
    "off",
  ],
  "@typescript-eslint/class-literal-property-style": [
    "off",
  ],
  "@typescript-eslint/comma-dangle": [
    "off",
  ],
  "@typescript-eslint/comma-spacing": [
    "off",
  ],
  "@typescript-eslint/consistent-generic-constructors": [
    "off",
  ],
  "@typescript-eslint/consistent-indexed-object-style": [
    "off",
  ],
  "@typescript-eslint/consistent-type-assertions": [
    "off",
  ],
  "@typescript-eslint/consistent-type-definitions": [
    "off",
  ],
  "@typescript-eslint/explicit-function-return-type": [
    "off",
  ],
  "@typescript-eslint/explicit-member-accessibility": [
    "error",
    {
      "overrides": {
        "accessors": "no-public",
        "constructors": "no-public",
        "parameterProperties": "no-public",
      },
    },
  ],
  "@typescript-eslint/explicit-module-boundary-types": [
    "off",
  ],
  "@typescript-eslint/func-call-spacing": [
    "error",
    "never",
  ],
  "@typescript-eslint/indent": [
    "off",
  ],
  "@typescript-eslint/key-spacing": [
    "off",
  ],
  "@typescript-eslint/keyword-spacing": [
    "off",
  ],
  "@typescript-eslint/lines-around-comment": [
    0,
  ],
  "@typescript-eslint/member-delimiter-style": [
    "off",
  ],
  "@typescript-eslint/no-array-constructor": [
    "error",
  ],
  "@typescript-eslint/no-confusing-non-null-assertion": [
    "off",
  ],
  "@typescript-eslint/no-duplicate-enum-values": [
    "error",
  ],
  "@typescript-eslint/no-empty-function": [
    "error",
  ],
  "@typescript-eslint/no-empty-interface": [
    "error",
  ],
  "@typescript-eslint/no-explicit-any": [
    "off",
  ],
  "@typescript-eslint/no-extra-non-null-assertion": [
    "error",
  ],
  "@typescript-eslint/no-extra-parens": [
    "off",
  ],
  "@typescript-eslint/no-extra-semi": [
    "off",
  ],
  "@typescript-eslint/no-floating-promises": [
    "error",
  ],
  "@typescript-eslint/no-inferrable-types": [
    "error",
  ],
  "@typescript-eslint/no-loss-of-precision": [
    "error",
  ],
  "@typescript-eslint/no-misused-new": [
    "error",
  ],
  "@typescript-eslint/no-namespace": [
    "error",
  ],
  "@typescript-eslint/no-non-null-asserted-optional-chain": [
    "error",
  ],
  "@typescript-eslint/no-non-null-assertion": [
    "warn",
  ],
  "@typescript-eslint/no-redeclare": [
    "error",
  ],
  "@typescript-eslint/no-this-alias": [
    "error",
  ],
  "@typescript-eslint/no-unnecessary-type-constraint": [
    "error",
  ],
  "@typescript-eslint/no-unsafe-declaration-merging": [
    "error",
  ],
  "@typescript-eslint/no-unused-vars": [
    "error",
    {
      "argsIgnorePattern": "^_",
      "ignoreRestSiblings": true,
      "varsIgnorePattern": "^_",
    },
  ],
  "@typescript-eslint/no-use-before-define": [
    "off",
  ],
  "@typescript-eslint/no-useless-constructor": [
    "off",
  ],
  "@typescript-eslint/no-var-requires": [
    "error",
  ],
  "@typescript-eslint/object-curly-spacing": [
    "off",
  ],
  "@typescript-eslint/prefer-as-const": [
    "error",
  ],
  "@typescript-eslint/prefer-for-of": [
    "off",
  ],
  "@typescript-eslint/prefer-function-type": [
    "off",
  ],
  "@typescript-eslint/prefer-namespace-keyword": [
    "error",
  ],
  "@typescript-eslint/quotes": [
    0,
  ],
  "@typescript-eslint/semi": [
    "error",
    "always",
  ],
  "@typescript-eslint/space-before-blocks": [
    "off",
  ],
  "@typescript-eslint/space-before-function-paren": [
    "off",
  ],
  "@typescript-eslint/space-infix-ops": [
    "off",
  ],
  "@typescript-eslint/triple-slash-reference": [
    "error",
  ],
  "@typescript-eslint/type-annotation-spacing": [
    "off",
  ],
  "array-bracket-newline": [
    "off",
  ],
  "array-bracket-spacing": [
    "off",
  ],
  "array-callback-return": [
    "off",
  ],
  "array-element-newline": [
    "off",
  ],
  "arrow-body-style": [
    "off",
  ],
  "arrow-parens": [
    "warn",
    "always",
  ],
  "arrow-spacing": [
    "off",
  ],
  "babel/object-curly-spacing": [
    "off",
  ],
  "babel/quotes": [
    0,
  ],
  "babel/semi": [
    "off",
  ],
  "block-spacing": [
    "off",
  ],
  "brace-style": [
    "off",
  ],
  "camelcase": [
    "error",
    {
      "ignoreDestructuring": true,
      "ignoreGlobals": false,
      "ignoreImports": false,
      "properties": "never",
    },
  ],
  "class-methods-use-this": [
    "off",
  ],
  "comma-dangle": [
    "error",
    {
      "arrays": "always-multiline",
      "exports": "always-multiline",
      "functions": "never",
      "imports": "always-multiline",
      "objects": "always-multiline",
    },
  ],
  "comma-spacing": [
    "off",
  ],
  "comma-style": [
    "off",
  ],
  "computed-property-spacing": [
    "off",
  ],
  "consistent-return": [
    "off",
  ],
  "constructor-super": [
    "error",
  ],
  "curly": [
    0,
  ],
  "dot-location": [
    "off",
  ],
  "dot-notation": [
    "warn",
  ],
  "eol-last": [
    "off",
  ],
  "eqeqeq": [
    "error",
    "smart",
  ],
  "flowtype/boolean-style": [
    "off",
  ],
  "flowtype/delimiter-dangle": [
    "off",
  ],
  "flowtype/generic-spacing": [
    "off",
  ],
  "flowtype/object-type-curly-spacing": [
    "off",
  ],
  "flowtype/object-type-delimiter": [
    "off",
  ],
  "flowtype/quotes": [
    "off",
  ],
  "flowtype/semi": [
    "off",
  ],
  "flowtype/space-after-type-colon": [
    "off",
  ],
  "flowtype/space-before-generic-bracket": [
    "off",
  ],
  "flowtype/space-before-type-colon": [
    "off",
  ],
  "flowtype/union-intersection-spacing": [
    "off",
  ],
  "for-direction": [
    "error",
  ],
  "func-call-spacing": [
    "off",
  ],
  "function-call-argument-newline": [
    "off",
  ],
  "function-paren-newline": [
    "off",
  ],
  "generator-star": [
    "off",
  ],
  "generator-star-spacing": [
    "off",
  ],
  "getter-return": [
    "error",
  ],
  "implicit-arrow-linebreak": [
    "off",
  ],
  "import/extensions": [
    "off",
  ],
  "import/first": [
    "warn",
  ],
  "import/no-extraneous-dependencies": [
    "off",
  ],
  "import/no-unresolved": [
    "off",
  ],
  "import/prefer-default-export": [
    "off",
  ],
  "indent": [
    "off",
  ],
  "indent-legacy": [
    "off",
  ],
  "jsx-a11y/alt-text": [
    "off",
  ],
  "jsx-a11y/label-has-for": [
    "off",
  ],
  "jsx-a11y/no-noninteractive-element-interactions": [
    "off",
  ],
  "jsx-a11y/no-static-element-interactions": [
    "off",
  ],
  "jsx-no-target-blank": [
    "off",
  ],
  "jsx-quotes": [
    "off",
  ],
  "key-spacing": [
    "off",
  ],
  "keyword-spacing": [
    "off",
  ],
  "linebreak-style": [
    "error",
    "unix",
  ],
  "lines-around-comment": [
    0,
  ],
  "max-len": [
    "off",
  ],
  "max-statements-per-line": [
    "off",
  ],
  "multiline-ternary": [
    "off",
  ],
  "new-cap": [
    "error",
    {
      "capIsNew": false,
      "newIsCap": true,
      "properties": true,
    },
  ],
  "new-parens": [
    "off",
  ],
  "newline-per-chained-call": [
    "off",
  ],
  "no-alert": [
    "off",
  ],
  "no-array-constructor": [
    "off",
  ],
  "no-arrow-condition": [
    "off",
  ],
  "no-case-declarations": [
    "error",
  ],
  "no-class-assign": [
    "error",
  ],
  "no-comma-dangle": [
    "off",
  ],
  "no-compare-neg-zero": [
    "error",
  ],
  "no-cond-assign": [
    "warn",
  ],
  "no-confusing-arrow": [
    0,
  ],
  "no-console": [
    "error",
  ],
  "no-const-assign": [
    "error",
  ],
  "no-constant-condition": [
    "error",
  ],
  "no-control-regex": [
    "error",
  ],
  "no-debugger": [
    "error",
  ],
  "no-delete-var": [
    "error",
  ],
  "no-dupe-args": [
    "error",
  ],
  "no-dupe-class-members": [
    "error",
  ],
  "no-dupe-keys": [
    "error",
  ],
  "no-duplicate-case": [
    "error",
  ],
  "no-else-return": [
    "warn",
  ],
  "no-empty": [
    "error",
  ],
  "no-empty-character-class": [
    "error",
  ],
  "no-empty-function": [
    "off",
  ],
  "no-empty-pattern": [
    "error",
  ],
  "no-ex-assign": [
    "error",
  ],
  "no-extra-boolean-cast": [
    "warn",
  ],
  "no-extra-parens": [
    "off",
  ],
  "no-extra-semi": [
    "off",
  ],
  "no-fallthrough": [
    "error",
  ],
  "no-floating-decimal": [
    "off",
  ],
  "no-func-assign": [
    "error",
  ],
  "no-global-assign": [
    "error",
  ],
  "no-import-assign": [
    "off",
  ],
  "no-inner-declarations": [
    "error",
  ],
  "no-invalid-regexp": [
    "error",
  ],
  "no-irregular-whitespace": [
    "error",
  ],
  "no-lonely-if": [
    "warn",
  ],
  "no-loss-of-precision": [
    "off",
  ],
  "no-mixed-operators": [
    "warn",
    {
      "allowSamePrecedence": true,
      "groups": [
        [
          "&",
          "|",
          "^",
          "~",
          "<<",
          ">>",
          ">>>",
        ],
        [
          "&&",
          "||",
        ],
      ],
    },
  ],
  "no-mixed-spaces-and-tabs": [
    "off",
  ],
  "no-multi-spaces": [
    "off",
  ],
  "no-multiple-empty-lines": [
    "off",
  ],
  "no-nested-ternary": [
    "off",
  ],
  "no-new-symbol": [
    "error",
  ],
  "no-obj-calls": [
    "error",
  ],
  "no-octal": [
    "error",
  ],
  "no-param-reassign": [
    "error",
  ],
  "no-plusplus": [
    "error",
    {
      "allowForLoopAfterthoughts": true,
    },
  ],
  "no-redeclare": [
    "off",
  ],
  "no-regex-spaces": [
    "error",
  ],
  "no-reserved-keys": [
    "off",
  ],
  "no-self-assign": [
    "error",
  ],
  "no-setter-return": [
    "off",
  ],
  "no-shadow": [
    "off",
  ],
  "no-space-before-semi": [
    "off",
  ],
  "no-spaced-func": [
    "off",
  ],
  "no-sparse-arrays": [
    "error",
  ],
  "no-tabs": [
    0,
  ],
  "no-this-before-super": [
    "error",
  ],
  "no-trailing-spaces": [
    "off",
  ],
  "no-undef": [
    "error",
  ],
  "no-underscore-dangle": [
    "off",
  ],
  "no-unexpected-multiline": [
    0,
  ],
  "no-unreachable": [
    "error",
  ],
  "no-unsafe-finally": [
    "error",
  ],
  "no-unsafe-negation": [
    "error",
  ],
  "no-unused-labels": [
    "error",
  ],
  "no-unused-vars": [
    "off",
  ],
  "no-use-before-define": [
    "off",
  ],
  "no-useless-constructor": [
    "off",
  ],
  "no-useless-escape": [
    "off",
  ],
  "no-var": [
    "error",
  ],
  "no-whitespace-before-property": [
    "off",
  ],
  "no-wrap-func": [
    "off",
  ],
  "nonblock-statement-body-position": [
    "off",
  ],
  "object-curly-newline": [
    "off",
  ],
  "object-curly-spacing": [
    "off",
  ],
  "object-property-newline": [
    "off",
  ],
  "object-shorthand": [
    "warn",
  ],
  "one-var-declaration-per-line": [
    "off",
  ],
  "operator-linebreak": [
    "off",
  ],
  "padded-blocks": [
    "off",
  ],
  "prefer-const": [
    "warn",
  ],
  "prefer-rest-params": [
    "error",
  ],
  "prefer-spread": [
    "error",
  ],
  "prefer-template": [
    "warn",
  ],
  "quote-props": [
    "warn",
    "as-needed",
  ],
  "quotes": [
    "off",
  ],
  "react/jsx-child-element-spacing": [
    "off",
  ],
  "react/jsx-closing-bracket-location": [
    "off",
  ],
  "react/jsx-closing-tag-location": [
    "off",
  ],
  "react/jsx-curly-newline": [
    "off",
  ],
  "react/jsx-curly-spacing": [
    "off",
  ],
  "react/jsx-equals-spacing": [
    "off",
  ],
  "react/jsx-first-prop-new-line": [
    "off",
  ],
  "react/jsx-indent": [
    "off",
  ],
  "react/jsx-indent-props": [
    "off",
  ],
  "react/jsx-max-props-per-line": [
    "off",
  ],
  "react/jsx-newline": [
    "off",
  ],
  "react/jsx-one-expression-per-line": [
    "off",
  ],
  "react/jsx-props-no-multi-spaces": [
    "off",
  ],
  "react/jsx-space-before-closing": [
    "off",
  ],
  "react/jsx-tag-spacing": [
    "off",
  ],
  "react/jsx-wrap-multilines": [
    "off",
  ],
  "require-yield": [
    "error",
  ],
  "rest-spread-spacing": [
    "off",
  ],
  "semi": [
    "off",
  ],
  "semi-spacing": [
    "off",
  ],
  "semi-style": [
    "off",
  ],
  "space-after-function-name": [
    "off",
  ],
  "space-after-keywords": [
    "off",
  ],
  "space-before-blocks": [
    "off",
  ],
  "space-before-function-paren": [
    "warn",
    {
      "anonymous": "always",
      "asyncArrow": "always",
      "named": "never",
    },
  ],
  "space-before-function-parentheses": [
    "off",
  ],
  "space-before-keywords": [
    "off",
  ],
  "space-in-brackets": [
    "off",
  ],
  "space-in-parens": [
    "off",
  ],
  "space-infix-ops": [
    "off",
  ],
  "space-return-throw-case": [
    "off",
  ],
  "space-unary-ops": [
    "off",
  ],
  "space-unary-word-ops": [
    "off",
  ],
  "standard/array-bracket-even-spacing": [
    "off",
  ],
  "standard/computed-property-even-spacing": [
    "off",
  ],
  "standard/object-curly-even-spacing": [
    "off",
  ],
  "switch-colon-spacing": [
    "off",
  ],
  "template-curly-spacing": [
    "off",
  ],
  "template-tag-spacing": [
    "off",
  ],
  "unicode-bom": [
    "off",
  ],
  "unicorn/empty-brace-spaces": [
    "off",
  ],
  "unicorn/no-nested-ternary": [
    "off",
  ],
  "unicorn/number-literal-case": [
    "off",
  ],
  "use-isnan": [
    "error",
  ],
  "use-macros/graphql-tag": [
    "error",
  ],
  "valid-typeof": [
    "error",
  ],
  "vue/array-bracket-newline": [
    "off",
  ],
  "vue/array-bracket-spacing": [
    "off",
  ],
  "vue/array-element-newline": [
    "off",
  ],
  "vue/arrow-spacing": [
    "off",
  ],
  "vue/block-spacing": [
    "off",
  ],
  "vue/block-tag-newline": [
    "off",
  ],
  "vue/brace-style": [
    "off",
  ],
  "vue/comma-dangle": [
    "off",
  ],
  "vue/comma-spacing": [
    "off",
  ],
  "vue/comma-style": [
    "off",
  ],
  "vue/dot-location": [
    "off",
  ],
  "vue/func-call-spacing": [
    "off",
  ],
  "vue/html-closing-bracket-newline": [
    "off",
  ],
  "vue/html-closing-bracket-spacing": [
    "off",
  ],
  "vue/html-end-tags": [
    "off",
  ],
  "vue/html-indent": [
    "off",
  ],
  "vue/html-quotes": [
    "off",
  ],
  "vue/html-self-closing": [
    0,
  ],
  "vue/key-spacing": [
    "off",
  ],
  "vue/keyword-spacing": [
    "off",
  ],
  "vue/max-attributes-per-line": [
    "off",
  ],
  "vue/max-len": [
    0,
  ],
  "vue/multiline-html-element-content-newline": [
    "off",
  ],
  "vue/multiline-ternary": [
    "off",
  ],
  "vue/mustache-interpolation-spacing": [
    "off",
  ],
  "vue/no-extra-parens": [
    "off",
  ],
  "vue/no-multi-spaces": [
    "off",
  ],
  "vue/no-spaces-around-equal-signs-in-attribute": [
    "off",
  ],
  "vue/object-curly-newline": [
    "off",
  ],
  "vue/object-curly-spacing": [
    "off",
  ],
  "vue/object-property-newline": [
    "off",
  ],
  "vue/operator-linebreak": [
    "off",
  ],
  "vue/quote-props": [
    "off",
  ],
  "vue/script-indent": [
    "off",
  ],
  "vue/singleline-html-element-content-newline": [
    "off",
  ],
  "vue/space-in-parens": [
    "off",
  ],
  "vue/space-infix-ops": [
    "off",
  ],
  "vue/space-unary-ops": [
    "off",
  ],
  "vue/template-curly-spacing": [
    "off",
  ],
  "wrap-iife": [
    "off",
  ],
  "wrap-regex": [
    "off",
  ],
  "yield-star-spacing": [
    "off",
  ],
}
`;

exports[`eslint-config-wantedly-typescript/without-react should match snapshot for: settings 1`] = `{}`;
